src/rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -243,6 +243,11 @@ fi
 mkdir -p "/tmp/pai-root/log/$APP_ID/$CONTAINER_ID"
 ln -s /tmp/pai-root/log/$APP_ID/$CONTAINER_ID/DockerContainerDebug.log $LAUNCHER_LOG_DIR/DockerContainerDebug.log
 
+# copy the hadoop configuration file
+hadoop_config_dir="hadoop_config"
+mkdir -p $hadoop_config_dir
```

Comment:
Do you need to remove it after the job is done?

Reply:
@fanyangCS I think we would better keep the configure files. First the configure file dir is in the yarn local folder. Yarn cleaner will delete it after a period of time. Currently in our setting it will be deleted after 86400 seconds. This allow us to debug the configure if the job fails. Secondly the configure file size is small, about 200K. Unlike the code, it doesn't take much space.

Reply:
Make sense. by the way, the current approach has one side effect: if we change the Hadoop config after the job launches, the Docker container's Hadoop will be outdated. If the Hadoop read failed, it may not retry since the system thinks this is user failures.
@mzmssg and @sterowang , should we handle this case?

Reply:
Another potential practice that we still mount the original hadoop config as read only, but we mount it into another path such as /etc/hadoop-dist, and in docker container script, we copy it to /etc/hadoop. By this way, in docker container, user would have possibilities to update hadoop config changes again if /etc/hadoop-dist get chance updated.

Reply:
> Another potential practice that we still mount the original hadoop config as read only, but we mount it into another path such as /etc/hadoop-dist, and in docker container script, we copy it to /etc/hadoop. By this way, in docker container, user would have possibilities to update hadoop config changes again if /etc/hadoop-dist get chance updated.

copy the configures to /etc/hadoop in docker container is hard for the user to debug the misconfigure problem when job fails. Since if the the configure is changed by the job and when it fails and exit, the configure files will lost. If we want the user job to sync the latest configure, we can further mount the host configure dir as readonly into the docker container as /etc/hadoop-configure-host. So user job can decide how to do the sync. But this is tricky. Do we need to support this?

Reply:
I think current solution is a little tricky(Copy from NM pod to host, then host to job container). Because NM config does only contain NM related fields, which may be not enough for users.

I'm not sure which field Spark on Yarn needed, if only RM address and HDFS address needed, I think we could pass them by ENVs, then users generate config by themselves.

Reply:
From the doc https://github.com/apache/spark/blob/master/docs/running-on-yarn.md . The configuration file directory is required to run spark on yarn. If we store a copy on hdfs. User still need to sync it if we change it.